test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs

```diff
@@ -42,18 +42,56 @@ public class CompatibilityTests
         [InlineData("net5.1", "netcoreapp3.1", true)]
         [InlineData("net6.0", "netcoreapp3.1", true)]
 
-        // net5.0-<osname> is not supported yet
-        [InlineData("net5.0-android", "net5.0", false)]
-        [InlineData("net5.0-android", "netcoreapp3.1", false)]
-        [InlineData("net5.0-ios", "net5.0", false)]
-        [InlineData("net5.0-macos", "net5.0", false)]
-        [InlineData("net5.0-tvos", "net5.0", false)]
-        [InlineData("net5.0-watchos", "net5.0", false)]
-        [InlineData("net5.0-windows", "net5.0", false)]
+        // net5.0-<osname> is compatible with net5.0 but not the inverse
+        [InlineData("net5.0-android", "net5.0", true)]
```

Comment:
as commented elsewhere, are TFMs with a platform identifier, but no platform version even supported/allowed?  How can NuGet do compat checks, when we don't know the version?

Reply:
They are indeed supported, and NuGetFrameworks without an OsVersion are perfectly valid -- they're evaluated later based on information we get from the SDK (that's tbd and it's one of the future things to do)

Reply:
This is the occasion where the NuGetFramework API needs to support one thing, but restore will actually require the min version :)

Reply:
I really don't want to delay this PR being merged, but just as Nikolche wrote, what needs to work from a user point of view working with their `csproj` file is not necesserily the same as what the `NuGetFramework` class needs to implement.

Another way to look at it:  I'm writing a custom desktop app that knows nothing about MSBuild or the .NET SDK, but uses `NuGet.Framework` from the package on nuget.org.  I want to check if two TFMs are compatible. If we don't enforce TPV being mandatory, how can we do the check?